examples/examples_with_aws/ReAct_Llama_2_Bedrock-WK.ipynb

```diff
@@ -0,0 +1,572 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def next_step(response):\n",
+    "    instruction = response[ : response.find('\\n- Observation:')]\n",
+    "    lines = instruction[instruction.rfind(\"Action:\"):].split(\"\\n\")\n",
+    "    action, action_input = lines[0].split(\": \")[1].strip(), lines[1].split(\": \")[1].strip()\n",
+    "    func = globals().get(action)\n",
+    "    observation = func(action_input)\n",
+    "    observation = observation[:observation[:350].rfind('. ')]\n",
+    "    return instruction + '\\n- Observation: ' + observation + '\\n- Thought:'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[INST] You are a useful AI agent. Answer the following questions as best you can. You have access to the following tools:\n",
+      "\n",
+      "Tools = ['duckduckgo_tool: Useful for when you need to search online about facts and events or retrieve news.', 'wikipedia_tool: Useful for when you need to answer general questions about people, places, companies, facts, historical events, or other subjects. Input should be a search query.', 'repl_tool: A Python shell. Use this to execute python commands or to calculate math expressions. Input should be a valid python command.']\n",
+      "\n",
+      "Use the following format:\n",
+      "\n",
+      "### Start\n",
+      "- Question: the input question you must answer\n",
+      "- Thought: explain your reasoning about what to do next\n",
+      "- Action: the action to take, should be one of ['duckduckgo_tool', 'wikipedia_tool', 'repl_tool']\n",
+      "- Action Input: the input to the action\n",
+      "- Observation: the result of the action\n",
+      "... (this Thought/Action/Action Input/Observation can repeat N times)\n",
+      "- Thought: I now know the final answer\n",
+      "- Final Answer: the final answer to the original input question\n",
+      "\n",
+      "Follow this format and Start! [/INST]\n",
+      "\n",
+      "### Start\n",
+      "- Question: What is Nvidia market cap?\n",
+      "- Thought:\u001b[32m\u001b[1m I don't know the current market cap of Nvidia, but I can search for it online.\n",
+      "- Action: duckduckgo_tool\n",
+      "- Action Input: \"Nvidia market cap\"\n",
+      "- Observation: Earnings Yield. 1.51%. PEGY Ratio. 0.1128. Operating PE Ratio. 59.62. Normalized PE Ratio. 66.07. In depth view into NVIDIA Market Cap including historical data from 1999, charts and stats. NVIDIA Market Cap. NVIDIA has a market cap or net worth of $1.98 trillion as of February 28, 2024. Its market cap has increased by 275.81% in one year\n",
+      "- Thought:\n"
+     ]
+    }
+   ],
+   "source": [
+    "response_observation = next_step(response)\n",
+    "new_query = query + '\\033[32m\\033[1m' + response_observation\n",
```

Comment:
maybe add a quick note saying '\\033[32m\\033[1m' contains escape code to set the text that follows to be bold green (another one later to be bold blue.